core/types/body_ext.go

```diff
@@ -0,0 +1,51 @@
+// (c) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package types
+
+import (
+	"github.com/ava-labs/libevm/rlp"
+)
+
+func GetBodyExtra(b *Body) *BodyExtra {
+	return extras.Body.Get(b)
+}
+
+func WithBodyExtra(b *Body, extra *BodyExtra) *Body {
+	extras.Body.Set(b, extra)
+	return b
+}
+
+// BodyExtra is a struct that contains extra fields used by Avalanche
+// in the body.
+// This type uses BodySerializable to encode and decode the extra fields
+// along with the upstream type for compatibility with existing network blocks.
+type BodyExtra struct {
+	Version uint32
+	ExtData *[]byte
+
+	// Fields removed from geth:
+	//  - withdrawals  Withdrawals
+}
+
+func (b *BodyExtra) RLPFieldsForEncoding(body *Body) *rlp.Fields {
+	return &rlp.Fields{
+		Required: []any{
+			body.Transactions,
+			body.Uncles,
+			b.Version,
+			b.ExtData,
+		},
+	}
+}
+
+func (b *BodyExtra) RLPFieldPointersForDecoding(body *Body) *rlp.Fields {
+	return &rlp.Fields{
+		Required: []any{
+			&body.Transactions,
+			&body.Uncles,
+			&b.Version,
+			&b.ExtData,
```

Comment:
Maybe a dumb remark, but note this doesn't use rlp.Nillable, I believe this is what we want correct?